subxt/src/blocks/extrinsic_types.rs

```diff
@@ -709,9 +728,12 @@ impl<'a> ExtrinsicSignedExtension<'a> {
 
     /// Signed Extension as a [`scale_value::Value`]
     pub fn value(&self) -> Result<DecodedValue, Error> {
-        let value =
-            DecodedValue::decode_as_type(&mut &self.bytes[..], self.ty_id, self.metadata.types())?;
+        self.as_type()
+    }
 
+    /// Decodes the `extra` bytes of this Signed Extension into a static type.
+    pub fn as_type<E: DecodeAsType>(&self) -> Result<E, Error> {
+        let value = E::decode_as_type(&mut &self.bytes[..], self.ty_id, self.metadata.types())?;
         Ok(value)
     }
```

Comment:
So for eg events we have `pub fn as_event<E: StaticEvent>(&self) -> Result<Option<E>, Error>`, and then for extrinsics we have `pub fn as_extrinsic<E: StaticExtrinsic>(&self) -> Result<Option<E>, Error>`.

I'd be inclined to add something here like:

```rust
pub fn as_signed_extra<S: SignedExtension>(&self) -> Result<Option<S::Decoded>, Error> {
    // Have sanity check that identifier lines up with `SignedExtension` given and then the 
    // usual decode logic, `Ok(None)` if name doesn't match (like the others).
}
```

This would be consistent with the above, and also be a little stricter than `as_type`, only allowing something that looks like the correct signed extension type to even be tried in the first place. I have no real objections to keeping `as_type` around too, but probably would remove it for now in favour of people defining and using more proper `SignedExtension` impls here :)